lms/djangoapps/instructor/tests/test_legacy_raw_download_csv.py

```diff
@@ -104,6 +107,147 @@ def test_grade_summary_data(self):
             ]
         }
 
+        non_grade_columns = 5
+        if (get_grades is False) or (use_offline is True):
+            expected_data["header"] = expected_data["header"][:non_grade_columns]
+            for index, rec in enumerate(expected_data["data"]):
+                expected_data["data"][index] = rec[:non_grade_columns]
+            if use_offline is True:
+                expected_data['assignments'] = []
+            if get_grades is False:
+                del expected_data['assignments']
+        elif get_raw_scores is True:
+            expected_data["data"] =  [
+                [
+                    1, u'u1', u'username', u'view@test.com',
+                    '', None, None, None
+                ],
+                [
+                    2, u'u2', u'username', u'view2@test.com',
+                    '', 0.0, 1.0, 0.0
+                ],
+            ]
+            expected_data["assignments"] = [u'p3', u'p2', u'p1']
+            expected_data["header"] = [
+                u'ID', u'Username', u'Full Name', u'edX email',
+                u'External email', u'p3', u'p2', u'p1'
+            ]
+
+        return expected_data
+
+    def test_grade_summary_data_defaults(self):
+        """
+        Test grade summary data report generation with all default kwargs.
+        """
+        request = DummyRequest()
+        self.answer_question()
+        data = get_student_grade_summary_data(request, self.course)
+        expected_data = self.get_expected_grade_data()
+
+        for key in ['assignments', 'header']:
+            self.assertListEqual(expected_data[key], data[key])
+
+        for index, student in enumerate(expected_data['students']):
+            self.assertEqual(
```

Comment:
Explain what these asserts are testing - think: if these tests fail, and the dev needing to fix them doesn't know this feature, how could you describe what you are testing in a way that makes sense. Your code should be self documenting and be able to stand on its own.